pkg/collector/containerlogs_containerd_collector.go

```diff
@@ -0,0 +1,160 @@
+package collector
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// ContainerLogsCollectorContainerD defines a ContainerLogs Collector struct for containerd clusters
+type ContainerLogsContainerDCollector struct {
+	data map[string]string
+}
+
+// NewContainerLogsCollectorContainerD is a constructor
+func NewContainerLogsContainerDCollector() *ContainerLogsContainerDCollector {
+	return &ContainerLogsContainerDCollector{
+		data: make(map[string]string),
+	}
+}
+
+func (collector *ContainerLogsContainerDCollector) GetName() string {
+	return "containerlogscontainerd"
+}
+
+type ContainerLog struct {
+	podname       string
+	namespace     string
+	containerName string
+	containeruid  string
+	filepath      string
+}
+
+type ContainerLogSelector struct {
+	namespace           string
+	containerNamePrefix string
+}
+
+//special selector which indicates logs from containers in all namespaces should be selected for collection
+const allNamespacesSelector = "--all-namespaces"
+
+// NOTE pod log files not currently used are in sub-directories of /var/log/pods and have format NAMESPACE_PODNAME_SOMEGUIDMEANINGTBD/CONTAINERNAME/#.log,
+// where I assume # is incremented and a new log created for each container restart (starts at # == 0)
+const containerLogDirectory = "/var/log/containers"
+
+func (collector *ContainerLogsContainerDCollector) GetData() map[string]string {
+	return collector.data
+}
+
+// Collect implements the interface method
+func (collector *ContainerLogsContainerDCollector) Collect() error {
+	selectorStrings := strings.Fields(os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST"))
```

Comment:
why this env variable?

I think any collector should collect all possible data.
What is done with (filtering, sorting, ...) might not be aks-periscope scope

Reply:
I disagree, collectors should not collect all possible data (although they should be able to). This was one of the original periscope requirements and AKS leverages this to provide a guarantee that when support runs periscope and examines the outputs in the process of working on a support case, we won't e.g. collect and examine data from customer namespaces which are running their workloads, and which might contain sensitive data

Reply:
💡Thoughts:  Might be good if shared a `os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST")` input will look like? , and to make things bit more clear. 

* What current impact this has? (I think all this is new build only specific to `container-d` work I reckon? which is separate functionality altogether or is this impacting the current container log mechanism) 
* What is the current behaviour this will effect/affect if-at-all it will? 
* What feature implication / impact will it have in existing use-case. (I think none, please do correct me)

Thanks. (above might help to see wider scope)

Reply:
this is needed to let this new collector match the existing behavior for the existing containerlogs collector (which this containerd collector is just a different implementation of). 

Examples of where it is defined:

https://github.com/Azure/aks-periscope/blob/2ea86f05fcfab1a77ea7f553e8801428633ff04c/deployment/aks-periscope.yaml#L127

https://github.com/Azure/aks-periscope/blob/2ea86f05fcfab1a77ea7f553e8801428633ff04c/deployment/config-map.yaml#L6

and of where it is referenced and used in the original docker based containerlog collector:

https://github.com/Azure/aks-periscope/blob/2ea86f05fcfab1a77ea7f553e8801428633ff04c/pkg/collector/containerlogs_collector.go#L28